src/sys/epoll.rs

```diff
@@ -102,37 +103,50 @@ impl EpollEvent {
 /// # }
 /// ```
 #[derive(Debug)]
-pub struct Epoll(pub OwnedFd);
-impl Epoll {
+#[repr(transparent)]
+pub struct Epoll<'fd> {
+    epoll_fd: OwnedFd,
+    _fds: std::marker::PhantomData<*mut &'fd ()>,
```

Comment:
We can not use `_fds: std::marker::PhantomData<BorrowedFd<'fd>>` here or the following code would compile without any issue:

```rs
use std::{ fs::File, os::fd::AsFd };
use nix::sys::epoll::{Epoll, EpollCreateFlags, EpollEvent};

fn main() {
    let file = File::open("Cargo.toml").unwrap();
    let epoll = Epoll::new(EpollCreateFlags::empty()).unwrap();
    epoll.add(file.as_fd(), EpollEvent::empty()).unwrap();

    drop(file);
    drop(epoll);
}
```

I haven't figured out the reason, but using an invariant type here would work

Reply:
Friendly ping @asomers, do you have any idea about this?